tests/port_dpb.py

```diff
@@ -0,0 +1,202 @@
+import redis
+import time
+import os
+import pytest
+from pytest import *
+import json
+import re
+
+class Port():
+    def __init__(self, dvs, name = None):
+        self._name = name
+        if name != None:
+            self._port_num = int(re.compile(r'(\d+)$').search(self._name).group(1)) 
+        self._alias = None
+        self._speed = None
+        self._lanes = []
+        self._index = None
+        self._lanes_db_str = None
+        self._lanes_asic_db_str = None
+        self._oid = None
+        self._dvs = dvs
+        self._cfg_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        self._cfg_db_ptbl = swsscommon.Table(self._cfg_db, "PORT")
+        self._app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+        self._app_db_ptbl = swsscommon.Table(self._app_db, swsscommon.APP_PORT_TABLE_NAME)
+        self._asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        self._asic_db_ptbl = swsscommon.Table(self._asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        #self._counter_db = swsscommon.DBConnector(swsscommon.COUNTERS_DB, dvs.redis_sock, 0)
+        #self._counter_db = redis.Redis(unix_socket_path=dvs.redis_sock, db=swsscommon.COUNTERS_DB)
+
+    def set_name(self, name):
+        self._name = name
+        self._port_num = int(re.compile(r'(\d+)$').search(self._name).group(1))
+
+    def set_speed(self, speed):
+        self._speed = speed
+
+    def set_alias(self, alias):
+        self._alias = alias
+
+    def set_lanes(self, lanes):
+        self._lanes = lanes
+        lanes_list = []
+        for lane in lanes:
+            lanes_list.append(int(lane))
+        lanes_list.sort()
+        self._lanes_db_str = str(lanes_list)[1:-1]
+        self._lanes_asic_db_str = str(len(lanes)) + ":" + self._lanes_db_str
+        self._lanes_asic_db_str = self._lanes_asic_db_str.replace(" ", "")
+
+    def set_index(self, index):
+        self._index = index
+
+    def get_speed(self):
+        return self._speed
+
+    def get_alias(self):
+        return self._alias
+
+    def get_lanes(self):
+        return self._lanes
+
+    def get_num_lanes(self):
+        return len(self._lanes)
+
+    def get_index(self):
+        return self._index
+
+    def get_name(self):
+        return self._name
+
+    def get_port_num(self):
+        return int(re.compile(r'(\d+)$').search(self._name).group(1))
+
+    def get_lanes_db_str(self):
+        return self._lanes_db_str
+
+    def get_lanes_asic_db_str(self):
+        return self._lanes_asic_db_str
+
+    def get_oid(self):
+        return self._oid
+
+    def print_port(self):
+        print "Port: %s Lanes: %s Speed: %d, Index: %d"%(self._name, self._lanes, self._speed, self._index) 
+
+    def port_merge(self, child_ports):
+        child_ports.sort(key=lambda x: x.get_port_num())
+        self.set_name(child_ports[0].get_name())
+        self.set_speed(child_ports[0].get_speed() * len(child_ports))
```

Comment:
This is not necessarily true.  The merge port could be at different speed than the sum of child ports.

Reply:
Thanks. Lets me add all speed of all child ports explicitly